tests/ops/service/connectors/test_saas_connector.py

```diff
@@ -878,6 +904,44 @@ def test_preferences_executable(
             mailchimp_transactional_connection_config_no_secrets.system_key: "complete"
         }, "Updated to skipped in graph task, not updated here"
 
+    def test_preferences_executable_notice_based_consent(
+        self,
+        db,
+        consent_policy,
+        privacy_request_with_consent_policy,
+        privacy_preference_history,
+        iterable_connection_config_no_secrets,
+    ):
+        name = iterable_connection_config_no_secrets.saas_config["type"]
+        register(name, SaaSRequestType.UPDATE_CONSENT)(valid_consent_update_override)
+        assert valid_consent_update_override == SaaSRequestOverrideFactory.get_override(
+            name, SaaSRequestType.UPDATE_CONSENT
+        )
+        privacy_preference_history.privacy_request_id = (
+            privacy_request_with_consent_policy.id
+        )
+        # todo- implement manually getting this key, remove this placeholder in test
+        privacy_preference_history.notice_key = "example_privacy_notice_1"
```

Comment:
Todo- need to confirm notice_key should not be auto-generated on the PrivacyPreferenceNoticeHistory model when a row on `PrivacyPreferenceNoticeHistory` is initially created.

Reply:
If we cannot expect it to be auto-generated, then we'll need to implement additional code to get notice_key based on the privacy preference history id. 

To do this:

* `privacy_notice_history_id` on `PrivacyPreferenceHistory` matches `privacy_notice_history_id` on one (or more?) of `privacy_notice.translations`

Reply:
update- from @galvana: 

> here's where the notice_key is set, it's in Fidesplus https://github.com/ethyca/fidesplus/blob/main/src/fidesplus/api/preferences/preferences_util.py#L444 but because this is a Fidesplus workflow, we can assume it will be there for us

Reply:
We should add a test in Fidesplus to assert that the `privacy_notice_history_id` val is set when notices served endpoint is hit, comment above it's for bi-directional consent